File: julia/Sandbox/src/Analysis/Filtering.jl
module Filtering

using DataFrames, Query

######################
## single condition ##
######################

# Query - stand alone
sweet_tooth_sa = person |>
    @filter(_.sweet_tooth=="yes") |>
    DataFrame

# Query - LINQ
sweet_tooth_q = @from i in person begin
    @where i.sweet_tooth=="yes"
    @select i
    @collect DataFrame
end


#########################
## multiple conditions ##
#########################

# Query - stand alone
sweet_tooth_older_sa = person |>
    @filter(_.sweet_tooth=="yes" && _.age > 20) |>
    DataFrame

# Query - LINQ
sweet_tooth_older_q = @from i in person begin
    @where i.sweet_tooth=="yes" && i.age > 20
    @select i
    @collect DataFrame
end


end # module

File: r/analysis/filtering.R
library(dplyr)

######################
## single condition ##
######################

# base
sweet_tooth_base <- person[person[["sweet_tooth"]] == "yes",]

# dplyr
sweet_tooth_dplyr <- person %>%
  filter(sweet_tooth == "yes")



#########################
## multiple conditions ##
#########################

# base
sweet_tooth_older_base <- person[person[["sweet_tooth"]] == "yes" &
                                 person[["age"]] > 20,]

# dplyr
sweet_tooth_older_dplyr <- person %>%
  filter(sweet_tooth == "yes",
         age > 20)
